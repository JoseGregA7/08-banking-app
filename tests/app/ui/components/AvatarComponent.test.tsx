import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test, vitest } from 'vitest';
import AvatarLogout from '../../../../src/app/ui/components/AvatarComponent';

describe('AvatarLogout', () => {
    const mockHandleEndSession = vitest.fn();

    test('should call handleEndSession when the avatar image is clicked', () => {
        const { getByTestId } = render(
            <AvatarLogout handleEndSession={mockHandleEndSession} />
        );

        fireEvent.click(getByTestId('avatar-logout-button'));

        expect(mockHandleEndSession).toHaveBeenCalledTimes(2);
    });

    test('should render the avatar image correctly', () => {
        const { getByTestId } = render(
            <AvatarLogout handleEndSession={mockHandleEndSession} />
        );

        const avatarImage = getByTestId('avatar-logout-button');
        expect(avatarImage).toBeInTheDocument();
        expect(avatarImage).toHaveAttribute('alt', 'Avatar de usuario - Cerrar sesiÃ³n');
    });

    test('should match snapshot', () => {
        const { asFragment } = render(
            <AvatarLogout handleEndSession={mockHandleEndSession} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
