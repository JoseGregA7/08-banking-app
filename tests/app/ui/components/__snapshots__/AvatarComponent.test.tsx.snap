// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarLogout > should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Cerrar sesión"
    class="avatar-logout"
    data-testid="avatar-logout-wrapper"
    role="button"
  >
    <img
      alt="Avatar de usuario - Cerrar sesión"
      class="avatar"
      data-testid="avatar-logout-button"
      role="presentation"
      src="avatar.webp"
    />
  </div>
</DocumentFragment>
`;
